datafusion/physical-expr/src/intervals/interval_aritmetic.rs

```diff
@@ -1730,20 +1757,23 @@ mod tests {
             ),
         ];
         for interval in intervals {
-            assert_eq!(interval.cardinality()?, distinct_f32);
+            assert_eq!(interval.cardinality()?.unwrap(), distinct_f32);
         }
 
+        // If the floating numbers has showned a homogeneous distribution pattern, the result would to be
```

Comment:
I don't understand this comment -- maybe @ozankabak  can double check too

Reply:
What @berkaysynnada is trying to say here is that the regular logarithmic distribution of floating-point numbers are only applicable outside of the -`(phi, phi)` interval where `phi` denotes the largest positive subnormal floating point number. In the latter interval, IEEE 754 floats are linearly distributed. So his quick back-of-the-envelope formula used to expect test results are not valid in ranges containing zero. Therefore, he manually supplies the actual expected cardinality.

Reply:
I have updated the comment to make it more understandable.